integration/ztoc_test.go

```diff
@@ -130,6 +164,221 @@ func TestSociZtocList(t *testing.T) {
 		}
 	})
 }
+func TestSociZtocInfo(t *testing.T) {
+	t.Parallel()
+	sh, done := newSnapshotterBaseShell(t)
+	defer done()
+	rebootContainerd(t, sh, "", "")
+
+	testImages := prepareSociIndices(t, sh)
+
+	getFullZtoc := func(sh *dockershell.Shell, ztocPath string) (*ztoc.Ztoc, error) {
+		output := sh.O("cat", ztocPath)
+		reader := bytes.NewReader(output)
+		z, err := ztoc.Unmarshal(reader)
+		return z, err
+	}
+
+	for _, img := range testImages {
+		img := img
```

Comment:
Why?

Reply:
This is something @turan18 and I discussed. Go's loops have a somewhat surprising behavior where the loop variable `img` is the same **instance** across all iterations. If you create a closure over it, you're closing over effectively a pointer which will be updated on the next loop. If you introduce `img := img`, you create a local variable that shadows the loop variable and is independent between executions.

The classic example is the following:

```
for _, n := range []int{1,2,3,4,5} {
    go func() {
        fmt.Println(n)
    }()
}

output:
5
5
5
5
5
```

There's a detailed explanation here:  https://go.dev/doc/faq#closures_and_goroutines


It's not a huge issue for us now because `t.Run` is synchronous by default. But if we tried to make the subtests run in parallel, we might only test the last test case in the list.

Reply:
When go routines are spawned from within a loop, they share the same reference to the thing being iterated, in this case the `img` value. When `t.Run` is called within a loop a go routine is spawned and each routine has a reference to the same value of `img`. To fix this you can reassign `img` to itself to make sure each routine has its own copy of img. Although, with these tests its not really needed as `img` is never directly referenced within t.Run, it is still good to have in cases where it may be referenced.

A better explanation here: https://appliedgo.com/blog/goroutine-closures

Reply:
Oops didn't see @Kern-- commented 😅